### DISCUSSION TITLE
How to tell where the completion candidates come from?

### DISCUSSION THREAD

**Discussion by amosbird** [company-mode/company-mode#1184] [Q&A] [answered]
Is there a way to display the actual used backend's name after each completion candidate?

**Comment by dgutov** [comment, marked as answer]
It's shown in the mode line.

As another option, [company-posframe](https://github.com/tumashu/company-posframe/) -- an alternative frontend -- has a feature where it displays the backend highlighted in an extra line at the botton of the popup.

**Comment by amosbird** [comment]
Hmm, I don't see it. Are there any specific settings to be turned on?

**Comment by dgutov** [comment]
Are you using vanilla Emacs?

**Comment by amosbird** [comment]
no, spacemacs.

**Comment by dgutov** [comment]
OK, then please see 

https://github.com/syl20bnr/spacemacs/#documentation
and
https://github.com/syl20bnr/spacemacs/#getting-help
